scripts/generate_benchmark_report.py

```
@@ -0,0 +1,381 @@
+            continue
+        # All instanovo versions kept separately for v1.1.2 vs v1.2.2 comparison
+        v122 = find_instanovo(entries, v122_label)
+        v112 = find_instanovo(entries, v112_label)
+        # For ranking: keep only best version per algorithm
+        best = best_per_algorithm(entries)
+        n_algos = len(best)
+        # rank by AUC desc
+        ranked = sorted(best, key=lambda r: -r[2])
+        # Find v1.2.2 rank — look for instanovo entry in best (v1.2.2 should be the best instanovo if it improved)
+        v122_rank = None
+        for i, (algo, ver, auc) in enumerate(ranked, start=1):
```

Ruff (0.15.14)

[warning] Loop control variable `auc` not used within loop body

(B007)